File: app/src/main/java/org/bspb/smartbirds/pro/prefs/HerptilePrefs.java
package org.bspb.smartbirds.pro.prefs;

import org.androidannotations.annotations.sharedpreferences.DefaultString;
import org.androidannotations.annotations.sharedpreferences.SharedPref;

/**
 * Created by dani on 14-11-18.
 */
@SharedPref
public interface HerptilePrefs {

    String herptileHabitat();

    @DefaultString("1")
    String herptileCount();


}

File: app/src/main/kotlin/org/bspb/smartbirds/pro/prefs/HerptilePrefs.kt
package org.bspb.smartbirds.pro.prefs

import android.content.Context
import android.content.SharedPreferences

class HerptilePrefs(context: Context) {

    companion object {
        const val KEY_HERPTILE_HABITAT = "herptileHabitat"
    }

    private var prefs: SharedPreferences = context.getSharedPreferences(
        (PrefsHelper.getLocalClassName(context) + "_HerptilePrefs"),
        0
    )

    fun getHerptileHabitat(): String {
        return prefs.getString(KEY_HERPTILE_HABITAT, "") ?: ""
    }

    fun setHerptileHabitat(value: String?) {
        prefs.edit()?.putString(KEY_HERPTILE_HABITAT, value)?.apply()
    }

}